src/ape/_compat.py

```diff
@@ -0,0 +1,13 @@
+import sys
+
+# We can remove this once we stop supporting python3.7.
+if sys.version_info >= (3, 8):
+    from functools import cached_property  # type: ignore
+    from functools import singledispatchmethod  # type: ignore
+    from importlib import metadata  # type: ignore
+    from typing import Literal
+else:
+    import importlib_metadata as metadata  # type: ignore
+    from backports.cached_property import cached_property  # type: ignore
+    from singledispatchmethod import singledispatchmethod  # type: ignore
+    from typing_extensions import Literal
```

Comment:
So `Literal` s not available until python 3.8 so that caused issues.

I don't think we want end-users importing compat methods from our `ape.utils` because they shouldn't need to. Keeping them here makes it easier to track (for deleting this code!) and makes it more protected rather than public.

Reply:
Definitely would like to see this as a separate PR

Reply:
No prob! https://github.com/ApeWorX/ape/pull/282